src/jit/lclvars.cpp

```diff
@@ -3687,63 +3685,66 @@ void Compiler::lvaMarkLclRefs(GenTree* tree, BasicBlock* block, GenTreeStmt* stm
         }
     }
 
-    /* Is this an assigment? */
-
-    if (tree->OperIsAssignment())
+    if (!isRecompute)
```

Comment:
Is it me or the only reason we need `isRecompute` is this dubious boolean optimization?!

Reply:
There is more down below -- setup for local assertion prop, checking/fixing local var types, etc.

It probably makes sense to revisit all the "first time only" stuff as it may not be needed or may be able to find better homes.